runtime/src/accounts_db.rs

```diff
@@ -2579,6 +2597,16 @@ impl AccountsDb {
             measure_shrink_all_candidates.as_ms() as usize
         );
         inc_new_counter_info!("shrink_all_candidate_slots-count", shrink_candidates_count);
+        let mut pended_counts: usize = 0;
+        if let Some(shrink_slots_next_batch) = shrink_slots_next_batch {
+            let mut shrink_slots = self.shrink_candidate_slots.lock().unwrap();
+            for (slot, stores) in shrink_slots_next_batch {
+                pended_counts += stores.len();
+                shrink_slots.entry(slot).or_default().extend(stores);
+            }
+        }
```

Comment:
Maybe a test for this as well that the stores are actually added back to the shrink candidates list and are triggered on the next shrink